Multithreading cannot build with Boost 1.64

With Boost 1.64, at least on Mac, Multithreading build fails with the following message:
```FAILED: applications/plugins/MultiThreading/CMakeFiles/MultiThreading.dir/src/TaskSchedulerBoost.cpp.o 
/usr/local/bin/ccache  clang++ -Qunused-arguments -Wno-deprecated-declarations  -DMultiThreading_EXPORTS -Iinclude -I../applications/plugins/MultiThreading/.. -isystem /usr/local/include -I../SofaKernel/SofaBase/../modules -I../SofaKernel/SofaSimulation/../modules -I../SofaKernel/framework/sofa/core/../.. -I../SofaKernel/framework/sofa/helper/../.. -F/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/System/Library/Frameworks -I../SofaKernel/extlibs/eigen-3.2.7 -ISofaKernel/SofaFramework -I../SofaKernel/extlibs/tinyxml -I../SofaKernel/framework/sofa/defaulttype/../.. -I../SofaMisc/../modules -I../SofaKernel/SofaCommon/../modules -isystem=/opt/local/include/Qt -Wall -W -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.11 -fPIC   -DSOFA_MULTITHREADING_PLUGIN -std=gnu++11 -MD -MT applications/plugins/MultiThreading/CMakeFiles/MultiThreading.dir/src/TaskSchedulerBoost.cpp.o -MF applications/plugins/MultiThreading/CMakeFiles/MultiThreading.dir/src/TaskSchedulerBoost.cpp.o.d -o applications/plugins/MultiThreading/CMakeFiles/MultiThreading.dir/src/TaskSchedulerBoost.cpp.o -c ../applications/plugins/MultiThreading/src/TaskSchedulerBoost.cpp
../applications/plugins/MultiThreading/src/TaskSchedulerBoost.cpp:192:18: error: overload resolution selected deleted operator '='
                        mTaskMutex.v_ = 0L;
                        ~~~~~~~~~~~~~ ^ ~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/atomic:1654:18: note: candidate function has been explicitly deleted
    atomic_flag& operator=(const atomic_flag&) = delete;
                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/atomic:1655:18: note: candidate function has been explicitly deleted
    atomic_flag& operator=(const atomic_flag&) volatile = delete;
                 ^
1 error generated.
```

This plugin should either be updated with the new boost API, or get rid of boost dependency.
I don't know this plugin enough...


______________________________________________________

Suggested labels: <!--- see https://github.com/sofa-framework/sofa/labels -->

Waiting for a fix, Multithreading and THMPGSpatialHashing plugins have been disabled in MacOS jobs.

Error output for THMPGSpatialHashing:
```
[214/302] Building CXX object applications/plugins/THMPGSpatialHashing/CMakeFiles/THMPGSpatialHashing.dir/THMPGHashTable.cpp.o
FAILED: applications/plugins/THMPGSpatialHashing/CMakeFiles/THMPGSpatialHashing.dir/THMPGHashTable.cpp.o 
/usr/local/bin/ccache  clang++ -Qunused-arguments -Wno-deprecated-declarations  -DTHMPGSpatialHashing_EXPORTS -I../SofaKernel/SofaBase/../modules -Iinclude -I../SofaKernel/SofaSimulation/../modules -I../SofaKernel/framework/sofa/core/../.. -I../SofaKernel/framework/sofa/helper/../.. -F/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/System/Library/Frameworks -isystem /usr/local/include -I../SofaKernel/extlibs/eigen-3.2.7 -ISofaKernel/SofaFramework -I../SofaKernel/extlibs/tinyxml -I../SofaKernel/framework/sofa/defaulttype/../.. -I../SofaKernel/SofaCommon/../modules -isystem=/opt/local/include/Qt -Wall -W -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.11 -fPIC   -DSOFA_BUILD_THMPGSPATIALHASHING -std=gnu++11 -MD -MT applications/plugins/THMPGSpatialHashing/CMakeFiles/THMPGSpatialHashing.dir/THMPGHashTable.cpp.o -MF applications/plugins/THMPGSpatialHashing/CMakeFiles/THMPGSpatialHashing.dir/THMPGHashTable.cpp.o.d -o applications/plugins/THMPGSpatialHashing/CMakeFiles/THMPGSpatialHashing.dir/THMPGHashTable.cpp.o -c ../applications/plugins/THMPGSpatialHashing/THMPGHashTable.cpp
In file included from ../applications/plugins/THMPGSpatialHashing/THMPGHashTable.cpp:1:
../applications/plugins/THMPGSpatialHashing/THMPGHashTable.h:5:10: fatal error: 'boost/unordered/detail/util.hpp' file not found
#include <boost/unordered/detail/util.hpp>
         ^
1 error generated.
```

--- a/applications/plugins/MultiThreading/src/TaskSchedulerBoost.cpp
+++ b/applications/plugins/MultiThreading/src/TaskSchedulerBoost.cpp
@@ -189,7 +189,6 @@ namespace sofa
 			mTaskCount		= 0;
 			mFinished		= false;
 			mCurrentStatus = NULL;
-			mTaskMutex.v_ = 0L;
 		}
 
 

--- a/applications/plugins/THMPGSpatialHashing/THMPGHashTable.h
+++ b/applications/plugins/THMPGSpatialHashing/THMPGHashTable.h
@@ -2,7 +2,12 @@
 
 #include <vector>
 #include <sofa/core/CollisionElement.h>
+#include <boost/version.hpp>
+#if BOOST_VERSION < 106400
 #include <boost/unordered/detail/util.hpp>
+#else
+#include <boost/unordered/detail/implementation.hpp>
+#endif
 #include <sofa/core/collision/Intersection.h>
 #include <sofa/core/collision/NarrowPhaseDetection.h>
 #include <sofa/helper/AdvancedTimer.h>

